Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1

```diff
@@ -81,6 +89,165 @@ function Add-DatabaseSecurity($userName, $role, $connString)
+    return $role
+}
+
+function Add-Group($authUrl, $name, $source, $accessToken){
+    $url = "$authUrl/groups"
+    $body = @{
+        id = "$name"
+        groupName = "$name"
+        groupSource = "$source"
+    }
+    return Invoke-Post $url $body $accessToken
+}
+
+function Add-User($authUrl, $name, $accessToken){
+    Write-Host "Adding User"
+    $url = "$authUrl/user"
+    $body = @{
+        subjectId = "$name"
+        identityProvider = "Windows"
+    }
+    return Invoke-Post $url $body $accessToken
+}
+
+function Add-RoleToUser($role, $user, $connString){
+    $query = "INSERT INTO RoleUsers
+              (CreatedBy, CreatedDateTimeUtc, RoleId, IdentityProvider, IsDeleted, SubjectId)
+              VALUES('fabric-installer', GetUtcDate(), @roleId, @identityProvider, 0, @subjectId)"
+
+    $roleId = $role.Id
+    $identityProvider = $user.identityProvider
+    $subjectId = $user.subjectId
+    Invoke-Sql $connString $query @{roleId=$roleId; identityProvider=$identityProvider; subjectId=$subjectId} | Out-Null
+}
+
+function Add-RoleToGroup($role, $group, $connString){
+    $query = "INSERT INTO GroupRoles
+              (CreatedBy, CreatedDateTimeUtc, GroupId, IsDeleted, RoleId)
+              VALUES('fabric-installer', GetUtcDate(), @groupId, 0, @roleId)"
+
+    $roleId = $role.Id
+    $groupId = $group.Id
+    Invoke-Sql $connString $query @{groupId=$groupId; roleId=$roleId;} | Out-Null
+}
+
+function Get-PrincipalContext($domain){
+    [System.Reflection.Assembly]::LoadWithPartialName("System.DirectoryServices.AccountManagement") | Out-Null
+    $ct = [System.DirectoryServices.AccountManagement.ContextType]::Domain 
+    $pc = New-Object System.DirectoryServices.AccountManagement.PrincipalContext -ArgumentList $ct, $domain
+    return $pc
+}
+
+function Test-IsUser($samAccountName, $domain){
+    $isUser = $false
+    $pc = Get-PrincipalContext -domain $domain
+    $user = [System.DirectoryServices.AccountManagement.UserPrincipal]::FindByIdentity($pc, $samAccountName)
+    if($user -ne $null){
+        $isUser = $true
+    }
+    return $isUser
+}
+
+function Test-IsGroup($samAccountName, $domain){
+    $isGroup = $false
+    $pc = Get-PrincipalContext -domain $domain
+    $group = [System.DirectoryServices.AccountManagement.GroupPrincipal]::FindByIdentity($pc, $samAccountName)
+    if($group -ne $null){
+        $isGroup = $true
+    }
+    return $isGroup
+}
+
+function Get-SamAccountFromAccountName($accountName){
+    $accountNameParts = $accountName.Split('\')
+    if($accountNameParts.Count -ne 2){
+        Write-Error "Please enter an account in the form DOMAIN\account. Halting installation." -ErrorAction Stop
+    }
+    $samAccountName = $accountNameParts[1]
+    return $samAccountName
+}
+
+function Add-AccountToDosAdminRole($accountName, $domain, $authorizationServiceURL, $accessToken, $connString){
+    $samAccountName = Get-SamAccountFromAccountName -accountName $accountName
```

Comment:
Should the variable passed to the `$authorizationServiceUrl` parameter be `$authorizationServiceURL` and not `$authorizationServiceUrl`?

Reply:
Yeah - I'll fix that up.